cli/src/utils/command.ts

```diff
@@ -0,0 +1,114 @@
+import { readFileSync } from 'fs';
+import { resolve } from 'path';
+import { URL } from 'url';
+import { Command } from '@oclif/command';
+import { CLIError } from '@oclif/errors';
+import * as Parser from '@oclif/parser';
+import open from 'open';
+import ms from 'ms';
+import terminalLink from 'terminal-link';
+import { LycanClient } from '@binaris/spice-node-client';
+import flags from './cli-flags';
+import userConfig from './user-config';
+
+const pjson = JSON.parse(readFileSync(resolve(__dirname, '../../package.json'), 'utf-8'));
+
+const LOGIN_PARAM = 'ticket';
+const TICKET_CLAIM_INTERVAL_MS = 1000;
+
+export default abstract class BaseCommand extends Command {
+  static cliBinName = pjson.oclif.bin as string;
+
+  protected realm?: string;
+  protected lycanClient?: LycanClient;
+
+  public static flags = {
+    help: flags.help({ char: 'h' }),
+    realm: flags.string({
```

Comment:
Can we drop this please?
`realm` is binaris internal.
I'd rather just have a base URL, it's much more generic.

Reply:
just to clarify, realm is not printed to the user, it's hidden and visible only inside the code
for example, do you want to ignore the realm environment variable completely?

Reply:
done